### ISSUE TITLE
defaults in xml conversion

### ISSUE THREAD

**Issue by andreasplesch** [castle-engine/demo-models#6]
Converting the example
https://github.com/castle-engine/demo-models/blob/master/prototypes/proto_events_test.x3dv

to xml encoding with view3dscene

leads to the integerKey field being omitted for IGenerate0:

```
                        <IntegerTrigger DEF="IGenerate0">
				<IS>
					<connect nodeField="triggerValue" protoField="int" />
				</IS>
			</IntegerTrigger>
```
The original x3dv file has a integerKey=0 explicit field value.
The xml encoding presumably omits the field value because the converter assumes that 0 is the default value. However, the default value is not defined in v3.2:

https://www.web3d.org/documents/specifications/19775-1/V3.2/Part01/components/utils.html#IntegerTrigger

and is -1 for versions 3.0 and 3.3.

So it is unexpected that 0 is used as a default value for version 3.2.

This is really converter issue, not a demo issue. Perhaps always including the field value if it is explicitly provided in the source file, and omitting it only if it is not given and has the same default value in the target x3d version might work.

Here the complete conversion:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE X3D PUBLIC "ISO//Web3D//DTD X3D 3.2//EN" "http://www.web3d.org/specifications/x3d-3.2.dtd">
<X3D profile="Interactive" version="3.2"
     xmlns:xsd="http://www.w3.org/2001/XMLSchema-instance"
     xsd:noNamespaceSchemaLocation="http://www.web3d.org/specifications/x3d-3.2.xsd">
<head>
	<meta name="generator" content="view3dscene, https://castle-engine.io/view3dscene.php" />
	<meta name="source" content="proto_events_test.x3dv" />
</head>
<Scene>
	<ProtoDeclare name="LightToggle">
		<ProtoInterface>
				<field accessType="inputOutput" type="SFBool" name="light_on" value="false" />
		</ProtoInterface>
		<ProtoBody>
			<Group>
				<Shape>
					<Appearance>
						<Material />
					</Appearance>
					<Sphere />
				</Shape>
				<DirectionalLight>
					<IS>
						<connect nodeField="on" protoField="light_on" />
					</IS>
				</DirectionalLight>
			</Group>
		</ProtoBody>
	</ProtoDeclare>
	<ProtoDeclare name="ConvertBoolToInt">
		<ProtoInterface>
				<field accessType="inputOnly" type="SFBool" name="bool" />
				<field accessType="outputOnly" type="SFInt32" name="int" />
		</ProtoInterface>
		<ProtoBody>
			<BooleanFilter DEF="BFilter">
				<IS>
					<connect nodeField="set_boolean" protoField="bool" />
				</IS>
			</BooleanFilter>
			<IntegerTrigger DEF="IGenerate0">
				<IS>
					<connect nodeField="triggerValue" protoField="int" />
				</IS>
			</IntegerTrigger>
			<IntegerTrigger DEF="IGenerate1"
				integerKey="1">
				<IS>
					<connect nodeField="triggerValue" protoField="int" />
				</IS>
			</IntegerTrigger>
			<ROUTE fromNode="BFilter" fromField="inputFalse" toNode="IGenerate0" toField="set_boolean" />
			<ROUTE fromNode="BFilter" fromField="inputTrue" toNode="IGenerate1" toField="set_boolean" />
		</ProtoBody>
	</ProtoDeclare>
	<ProtoInstance name="LightToggle" DEF="LT">
	</ProtoInstance>
	<KeySensor DEF="K" />
	<ROUTE fromNode="K" fromField="controlKey" toNode="LT" toField="set_light_on" />
	<Transform
		translation="0 -2 0">
		<DirectionalLight />
		<Switch DEF="LightOnVisualize"
			whichChoice="0">
			<Shape>
				<Appearance DEF="ATextLighOn">
					<Material />
				</Appearance>
				<Text
					string='"Hold Control key for the light." "Light is currently: OFF"' />
			</Shape>
			<Shape>
				<Appearance DEF="ATextLighOn">
					<Material />
				</Appearance>
				<Text
					string='"Hold Control key for the light." "Light is currently: ON"' />
			</Shape>
		</Switch>
	</Transform>
	<ProtoInstance name="ConvertBoolToInt" DEF="BToInt">
	</ProtoInstance>
	<ROUTE fromNode="LT" fromField="light_on_changed" toNode="BToInt" toField="bool" />
	<ROUTE fromNode="BToInt" fromField="int" toNode="LightOnVisualize" toField="set_whichChoice" />
</Scene>
</X3D>
```

**Comment by michaliskambi** [comment]
Thank you -- so we just had a wrong value for the default `IntegerTrigger.integerKey` . Looking at implementation, I had this comment:

```pascal
  { X3D specification mistakenly omitted default value for this field...
    I guess it's 0? }
  FFdIntegerKey := TSFInt32.Create(Self, true, 'integerKey', 0);
  AddField(FFdIntegerKey);
```

So I just fixed the default to be -1.

In CGE, after parsing the X3D file, we have to way of telling a difference between _"this field was explicitly specified in the file, with the value that equals default"_ and _"this field was not specified in the file"_. This is deliberate -- as I understand, the entire X3D standard is written in such way that it shouldn't matter. So the saving process has only a logic "do not write this field it it was equal to default".

I fixed it in Castle Game Engine, the view3dscene should be automatically updated by Jenkins after some time :)
